pkg/controllers/updaterun/initialization.go

```diff
@@ -380,13 +381,19 @@ func (r *Reconciler) computeRunStageStatus(
 	// Check if the clusters are all placed.
 	if len(allPlacedClusters) != len(allSelectedClusters) {
 		missingErr := controller.NewUserError(fmt.Errorf("some clusters are not placed in any stage"))
+		missingClusters := make([]string, 0, len(allSelectedClusters)-len(allPlacedClusters))
 		for cluster := range allSelectedClusters {
 			if _, ok := allPlacedClusters[cluster]; !ok {
-				klog.ErrorS(missingErr, "Cluster is missing in any stage", "cluster", cluster, "clusterStagedUpdateStrategy", updateStrategyName, "clusterStagedUpdateRun", updateRunRef)
+				missingClusters = append(missingClusters, cluster)
 			}
 		}
+		// Sort the missing clusters by their names to generate a stable error message.
+		sort.Strings(missingClusters)
+		klog.ErrorS(missingErr, "Clusters are missing in any stage", "clusters", strings.Join(missingClusters, ", "), "clusterStagedUpdateStrategy", updateStrategyName, "clusterStagedUpdateRun", updateRunRef)
+		// Only show the first 10 missing clusters in the CRP status.
+		missingClusters = missingClusters[:min(10, len(missingClusters))]
 		// no more retries here.
-		return fmt.Errorf("%w: %s", errInitializedFailed, missingErr.Error())
+		return fmt.Errorf("%w: %s (showing up to 10): %s", errInitializedFailed, missingErr.Error(), strings.Join(missingClusters, ", "))
```

Comment:
can we print out the total missing clusters even if we only return 10? I assume this error msg will be surfaced to the end user.

Reply:
added.